### PR TITLE
fix(landing-page):bugfix

### PR DESCRIPTION
**Pull request by Ahmedjaziri3**
## What does this do?

This PR addresses a UI rendering issue in iOS for a React Native application. It updates the `GradientBackground` component to ensure that styles are correctly applied, which solves the problem where the gradient background was not displaying as expected. The changes include modifying the usage of the `LinearGradient` component to properly handle the `className` prop by applying styles directly.

## Note:

This Update will be followed with multiple bugs fix updates before review 

## Why did you do this?

The changes were made to fix a rendering issue on iOS devices where the gradient background was not appearing correctly. By ensuring that the styles are applied properly, the UI now displays as intended, providing a consistent user experience across different platforms.

## Who/what does this impact?

These changes impact the `GradientBackground` component, which may be used in multiple places within the application. It's important for other developers who are working on UI components that use this gradient background to be aware of the changes. The changes should not have any adverse side effects but will enhance the visual consistency on iOS devices.

## How did you test this?

The changes were tested by running the application on both iOS and Android devices to ensure the gradient background renders correctly on all platforms. Visual inspections were performed to confirm that the styles are applied as expected and that the UI elements display properly without any layout issues.

## images reflect the first change:

image

image

image

### CODE REVIEW

**Comment by Ahmedjaziri3** [comment]
## What does this do?
Updates the landing page to set a default background color for the button. This ensures consistent button appearance and avoids color conflicts with different users' dynamic themes or system settings mainly it happens in the ios.

## images reflect the first change:

## before:

Capture d’écran 2024-07-31 à 12 09 27 PM

## after:

Capture d’écran 2024-07-31 à 12 05 42 PM

**Comment by Ahmedjaziri3** [comment]
## before:

Capture d’écran 2024-07-29 à 12 50 37 PM

Capture d’écran 2024-07-29 à 12 51 14 PM

Capture d’écran 2024-07-29 à 12 51 24 PM

Capture d’écran 2024-07-29 à 12 50 27 PM

Capture d’écran 2024-07-29 à 12 36 03 PM 2

Capture d’écran 2024-07-29 à 12 52 18 PM

Capture d’écran 2024-07-29 à 12 51 42 PM

Capture d’écran 2024-07-29 à 12 51 48 PM


Capture d’écran 2024-07-29 à 2 05 41 PM


## After:

Capture d’écran 2024-07-31 à 10 26 44 PM

Capture d’écran 2024-07-31 à 10 28 23 PM

Capture d’écran 2024-07-31 à 10 29 43 PM


Capture d’écran 2024-08-01 à 11 55 06 AM

Capture d’écran 2024-08-01 à 11 56 54 AM

## Before:

Capture d’écran 2024-07-31 à 10 31 28 PM

## After:

Capture d’écran 2024-08-01 à 11 31 53 AM

File: src/shared/components/gradient-background.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
 import React from 'react';
-import type { ViewProps } from 'react-native';
+import { StyleSheet, ViewStyle, StyleProp } from 'react-native';
 import LinearGradient from 'react-native-linear-gradient';
-
 import { colors } from '@/shared/components';
 
-type GradientBackgroundProps = ViewProps & {
+type GradientBackgroundProps = {
```
Comment (khalil7898) [inline src/shared/components/gradient-background.tsx:6]: emove the classeName from props typing for impacted component   and add nice commne texplaining the why with PR link

**Review by khalil7898 — CHANGES_REQUESTED** [review]
emove the classeName from props typing for impacted component and add nice commne texplaining the why with PR link

File: src/shared/components/header-title.tsx [hunk 2]
```diff
@@ -10,20 +10,35 @@ interface HeaderTitleProps {
...
-
+//the GradientBackground component class<name was removed and that styles are correctly applied, which solves the problem where the gradient background was not displaying as expected in IOS.
 export function HeaderTitle({ text, type }: HeaderTitleProps) {
   return (
     <>
       {type === 'custom' ? (
-        <GradientBackground className="flex h-[13%] w-full items-center justify-end bg-white pb-6">
-          <Text className="text-2xl font-bold text-primary-txt ">
+        <GradientBackground
+          style={{
```
Comment (MohamedMegdich1) [inline src/shared/components/header-title.tsx:19]: Why not use a StyleSheet instead?

File: src/shared/components/header-title.tsx [hunk 3]
```diff
@@ -10,21 +10,28 @@ interface HeaderTitleProps {
...
+            display: 'flex',
+            height: '13%',
+            width: '100%',
+            alignItems: 'center',
+            justifyContent: 'flex-end',
+            backgroundColor: 'white',
+          }}
+        >
+          <Text
+            style={{
```
[the remark this answers is not in the document]
Comment (Ahmedjaziri3) [reply src/shared/components/header-title.tsx]: In some cases, using StyleSheet.create for specific styles like marginTop: 40 can make the code more consistent and readable, especially if such a style doesnt showed up in ios when i use mt with tailwind

**Pull request merged**